src/huntsman/pocs/camera/camera.py

```diff
@@ -0,0 +1,187 @@
+import os
+import threading
+import time
+from contextlib import suppress
+from panoptes.utils.time import current_time
+from panoptes.pocs.camera.camera import AbstractCamera
+
+from huntsman.pocs.camera.utils import tune_exposure_time
+
+
+class AbstractHuntsmanCamera(AbstractCamera):
+
+    def take_observation(self, observation, headers=None, filename=None, blocking=False, **kwargs):
+        """Take an observation. Override of `panoptes.pocs.camera.camera.take_observation()` to
+        allow multifilter observations and dynamic exposure time tuning.
+
+        Gathers various header information, sets the file path, and calls
+            `take_exposure`. Also creates a `threading.Event` object and a
+            `threading.Thread` object. The Thread calls `process_exposure`
+            after the exposure had completed and the Event is set once
+            `process_exposure` finishes.
+        Args:
+            observation (~panoptes.pocs.scheduler.observation.Observation): Object
+                describing the observation
+            headers (dict, optional): Header data to be saved along with the file.
+            filename (str, optional): pass a filename for the output FITS file to
+                override the default file naming system.
+            blocking (bool): If method should wait for observation event to be complete
+                before returning, default False.
+            **kwargs (dict): Optional keyword arguments (`exptime`, dark)
+        Returns:
+            threading.Event: An event to be set when the image is done processing
+        """
+        observation_event = threading.Event()
+        # Setup the observation
+        exptime, file_path, image_id, metadata = self._setup_observation(observation,
+                                                                         headers,
+                                                                         filename,
+                                                                         **kwargs)
+
+        # pop exptime from kwarg as its now in exptime
+        exptime = kwargs.pop('exptime', observation.exptime.value)
```

Comment:
`exptime = kwargs.pop('exptime', observation.exptime.to_value("s"))`

Reply:
I think this is where the current test error is coming from. The argument name is (annoyingly) `seconds` instead of `exptime` here - that is why multiple `seconds` values get parsed into `take_exposure` (maybe)